R/mobr.R

```diff
@@ -57,290 +59,218 @@ make_comm_obj = function(x, plot_attr, binary=FALSE) {
+    interp_S_samp = pchip(c(1, rescaled_effort), c(1, S_samp), inds)
+    S_indiv = rarefaction(comm, 'indiv', inds)
+    deltaS = interp_S_samp - S_indiv
+    out = data.frame(inds = inds, deltaS = deltaS)
+    return(out)
 }
 
-# Auxillary function: spatially-explicit sample-based rarefaction 
-rarefy_sample_explicit = function(comm_one_group, xy_one_group) {
-  #plot_grp = dat_plot[dat_plot$group == group, ] 
-  #sp_grp = dat_sp[dat_plot$group == group, ]
-  row.names(comm_one_group) = as.character(seq(nrow(comm_one_group)))
-  explicit_loop = matrix(0, nrow(comm_one_group), nrow(comm_one_group))
-  pair_dist = as.matrix(dist(xy_one_group))
-  for (i in 1:nrow(comm_one_group)) {
-    focal_site = row.names(comm_one_group)[i]
-    dist_to_site = pair_dist[i, ]
-    # Shuffle plots, so that tied grouping is not biased by original order.
-    new_order = sample(1:nrow(comm_one_group))  
-    plots_new = row.names(comm_one_group)[new_order]
-    dist_new = dist_to_site[new_order]
-    plots_new_ordered = plots_new[order(dist_new)]
-    # Move focal site to the front
-    plots_new_ordered = c(focal_site, 
-                          plots_new_ordered[plots_new_ordered != focal_site])  
-    comm_ordered = comm_one_group[match(row.names(comm_one_group), plots_new_ordered), ]
-    # 1 for absence, 0 for presence
-    comm_bool = as.data.frame((comm_ordered == 0) * 1) 
-    rich = cumprod(comm_bool)
-    explicit_loop[ , i] = as.numeric(ncol(comm_one_group) - rowSums(rich))
-  }
-  explicit_S = apply(explicit_loop, 1, mean)
-  return(explicit_S)
-}
 
 #' Permute community matrix within groups
 #' 
-#' Two types of permutation can be carried out: 
-#' 1) each individual of each species is reassigned a plot randomly which removes 
-#' any patterns due to within and between plot spatial aggregation, but maintains
-#' species group abundance and therefore observed group N,
-#' 2) the total number of individuals in a plot is shuffled and then that many
-#' individuals are drawn randomly from the group specific species-abundance
-#' distribution for each plot which provides a mean of removing group differenes
-#' in the total number of individuals. 
+#' Two types of permutation can be carried out: 1) 'noagg': each individual of
+#' each species is reassigned a plot randomly which removes any patterns due to
+#' within and between plot spatial aggregation, but maintains species group
+#' abundance and therefore observed group N, and 2) 'swapN': the total number of
+#' individuals in a plot is shuffled and then that many individuals are drawn
+#' randomly from the group specific species-abundance distribution for each plot
+#' which provides a means of removing group differenes in the total number of
+#' individuals.
 #' 
-#' @param comm a site-by-species matrix
-#' @param swap either 'noagg' or 'swapN' 
+#' @param comm community matrix with plots as rows and species columns.
```

Comment:
yea the number of individuals of a given species can change with `swapN` but not with `noagg`. I can add that in as a note.